### ISSUE TITLE
not all stacks are displayed

### ISSUE THREAD

**Issue by andruwa13** [rancher/rancher#6445]
![rancher 2016-10-29 19-05-23](https://cloud.githubusercontent.com/assets/5846351/19831125/349a2e80-9e0b-11e6-9088-5663dac65280.png)

![rancher 2016-10-29 19-07-23](https://cloud.githubusercontent.com/assets/5846351/19831131/3b52bf58-9e0b-11e6-8d71-575436fb4f8c.png)

---

| Useful | Info |
| :-- | :-- |
| Versions | Rancher `v1.2.0-pre4-rc6` Cattle: `v0.171.0` UI: `v1.2.19` |
| Access | `localauth` `admin` |
| Orchestration | `Cattle` |
| Route | `catalog-tab.index` |

**Comment by andruwa13** [comment]
to release **v1.2.0-pre3** all works, v1.2.0-pre4-rc6 not work

**Comment by vincent99** [comment]
`linkedServices` isn't pulling in the name of the linked service as the default name when no "as name" is provided, so they all get mapped to the empty-string key and only the last one in wins:

```
Link to -> As Name
"one"   -> ""
"two"   -> ""
"three" -> "three"
"four"  -> "four"
```

Produces 

```json
"linkedServices": {
  "": "1s2",
  "three": "1s3",
  "four": "1s4",
},

**Comment by sangeethah** [comment]
Tested with latest build from master:
rancher/dns:v0.10.4
rancher/healthcheck:v0.1.0
rancher/net:v0.7.0
rancher/metadata:v0.6.3
rancher/plugin-manager:v0.2.1
rancher/scheduler:v0.1.2

Created a service with following yml file having external links with no name

```
version: '2'
services:
  testlinks:
    external_links:
    - testme/hello1
    - testme/hello2
    - testme/hello
    labels:
      io.rancher.container.pull_image: always
    tty: true
    image: sangeetha/testclient
    stdin_open: true
```
Service gets created with links to all the 3 services but the links get named as below:

<img width="1134" alt="screen shot 2016-11-16 at 1 49 43 pm" src="https://cloud.githubusercontent.com/assets/4266958/20367250/91d05740-ac03-11e6-9a52-4dde95b8d3f6.png">


@alena1108: Is this expected?

**Comment by alena1108** [comment]
@sangeethah this is fine

**Comment by sangeethah** [comment]
@alena1108 , Thanks.
@andruwa13 ,  fix for this issue will be available in 1.2 release.
